core/src/storage/impls/state.rs

```diff
@@ -239,6 +239,34 @@ impl StateTrait for State {
         self.get_from_all_tries(access_key, true)
     }
 
+    fn get_merkle_hash_wo_compressed_path(
+        &self, access_key: StorageKey,
+    ) -> Result<Option<MerkleHash>> {
+        // Get won't create any new nodes so it's fine to pass an empty
+        // owned_node_set.
+        let mut empty_owned_node_set: Option<OwnedNodeSet> =
+            Some(Default::default());
+
+        // TODO(thegaram): get from intermediate or snapshot if necessary
```

Comment:
We have to combine the merkle from snapshot and intermediate in order to capture contents in the storage.

Now there will be two corner cases:
1) there is no exact node for the storage root in intermediate or delta MPT. Solution a): When setting a StorageKey, we may want to set the StorageLayout again in the delta mpt / intermediate mpt; Solution b): when we compute the storage root, if there isn't any key value under storage root, it's simply MERKLE_NULL_NODE, when we stop on the way of a compressed_path, we call compute_node_merkle() on a virtually constructed StorageRoot node, taken the unfinished part of the compressed path as its children.

2) after conflux makes a snapshot, from the client side, the storage root seems to have changed. But we can send a flag (or snapshot height) along the way, to let client know that in this epoch, snapshotting happened, so that it can check if the current intermediate storage root matches the previous delta storage root, and check if the current delta storage root equals MERKLE_NULL_NODE.

These work can be done in the next PR.